src/BenchmarksServer/Startup.cs

```diff
@@ -560,13 +619,40 @@ private static string GetDotNetExecutable(string dotnetHome)
 
                     if (job.State == ServerState.Starting && e.Data.Contains("Application started"))
                     {
-                        job.State = ServerState.Running;
-                        job.Url = ComputeServerUrl(hostname, job.Scheme, job.Scenario);
                         Log.WriteLine($"Running job '{job.Id}' with scenario '{job.Scenario}'");
+                        job.Url = ComputeServerUrl(hostname, job.Scheme, job.Scenario);
+                        Log.WriteLine("Measuring startup time");
+                        
+                        using(var response = _httpClient.GetAsync(job.Url).GetAwaiter().GetResult())
+                        {
+                            var responseContent = response.Content.ReadAsStringAsync().GetAwaiter().GetResult();
+                            job.Startup = stopwatch.Elapsed;
+                        }
+
+                        // Measure a low-load latency by getting 5 samples synchronously
+                        // This could be done during the Client job but we are already measuring the Startup time here.
+                        const int LatencySamples = 10;
+                        var latencies = new List<TimeSpan>();
+                        for (var i = 0; i < LatencySamples; i++)
+                        {
+                            stopwatch.Restart();
+                            using(var response = _httpClient.GetAsync(job.Url).GetAwaiter().GetResult())
```

Comment:
Could `Task.GetAwaiter().GetResult()` be replaced by just `Task.Result`?

Reply:
Even better, `ProcessJobs()` is already async, so I think `StartProcess()` should also be made async, then you can just `await` rather than `Task.Result`.

Reply:
`Task.Result` returns exception wrapped in an AggregateException, `Task.GetAwaiter().GetResult()` returns actual exception